### PR TITLE
add new rotation and mixer for kd0aij's Caipirinha vtol

### PR DESCRIPTION
**Pull request by kd0aij**
**Purpose of this PR is only to discuss best practices for modifying mixers and the coding style of rotation_3f.**
I added a new autostart ID and mixer in order to store my settings in my fork, but would appreciate advice on better ways to go about this.
I also ran astyle on rotation.cpp and changed the multiple returns to breaks when adding my new case, but perhaps there was a good reason for the original style?

### CODE REVIEW

**Comment by LorenzMeier** [comment]
This all looks good, assuming you fixed the previously broken style. I need to do another pass tomorrow when I'm awake, not half-asleep like now.
Thanks!

File: ROMFS/px4fmu_common/mixers/caipirinha_vtol_ailrev.main.mix [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#!nsh
+# Caipirinha vtol mixer for PX4FMU
+#
+#===========================
+R: 2- 10000 10000 10000 0
+
+#mixer for the elevons
+M: 2
+O:      10000  10000      0 -10000  10000
+S: 1 0  -10000  -10000      0 -10000  10000
```
Comment (tumbili) [inline ROMFS/px4fmu_common/mixers/caipirinha_vtol_ailrev.main.mix]: Maybe we can format this to look a bit nicer?

Comment (kd0aij) [inline ROMFS/px4fmu_common/mixers/caipirinha_vtol_ailrev.main.mix]: I don't think it's a good idea to have a separate autostart ID and mixer for my particular configuration in the main repo, but haven't decided whether I want to override the default mixer by maintaining my own mixer on an SD card.

**Comment by LorenzMeier** [comment]
@kd0aij Is what you're trying to fix here just the servo travel direction? If yes, do we need a param for that?

**Comment by kd0aij** [comment]
@LorenzMeier Yes, my servos are apparently reversed compared to Roman's. But I also needed to reduce the maximum throw with my setup. I would like to have parameters for direction, travel and trim for each servo. This thread is also relevant: https://groups.google.com/d/msg/px4users/QdXbP_Q9y9s/H345rpacvoMJ

**Comment by kd0aij** [comment]
Not a question; just pointing out that the AP orientation is not specified in this PR. Although I suppose it could be set in the init.d file for ID 13003. 
The purpose of this PR is to get advice on best practice to fully document the configuration for a particular aircraft. How do you go about it? Are autostart IDs intended only for generic vehicles?

**Comment by bansiesta** [comment]
I suppose if this is the usual way the Pixhawk is mounted on the Caipi, than adding the orientation param to the autoconf is reasonable.

**Comment by kd0aij** [comment]
But it isn't. This is just how it happens to be mounted on my particular Caipi_vtol. Same for the mixer, too, especially when you consider the output scaling.

**Comment by bansiesta** [comment]
Ok, well then let's not include it. But I guess a different SYS_AUTOSTART id and different mixer file for "reversed" servos is ok.

**Comment by LorenzMeier** [comment]
@bansiesta Do you have any brilliant ids for the handling of reversed servos? I was thinking about MAVLink FTP for custom mixers or on the opposite a param for reversing servos. But how do we handle ESCs then?

**Comment by bansiesta** [comment]
+1 for a param for reversed servos!
And what's with the ESCs? What about a param for the min value?

**Pull request closed without merging**